InstanceManager.js

```diff
@@ -268,37 +269,52 @@ class InstanceManager {
       compactionKeep = process.env.AGENCY_COMPACTION_KEEP;
     }
     for (var i = 0; i < size; i++) {
-      instances.push(await this.runner
-          .createEndpoint()
-          .then(endpoint => {
-            const args = [
-              '--agency.activate=true',
-              '--agency.size=' + size,
-              '--agency.pool-size=' + size,
-              '--agency.wait-for-sync=true',
-              '--agency.supervision=true',
-              '--server.threads=16',
-              '--agency.supervision-frequency=0.5',
-              '--agency.supervision-grace-period=2.5',
-              '--agency.compaction-step-size='+compactionStep,
-              '--agency.compaction-keep-size='+compactionKeep,
-              '--agency.my-address=' + endpoint,
-              instances.length ?
-                `--agency.endpoint=${instances[0].endpoint}` : `--agency.endpoint=${endpoint}`
-            ];
-
-            this.agentCounter++;
-            return this.startArango(
-              'agency-' + this.agentCounter,
-              endpoint,
-              'agent',
-              args
-            );
-          }));
+      const endpoint = await this.runner.createEndpoint();
+      const args = [
+        '--agency.activate=true',
+        '--agency.size=' + size,
+        '--agency.pool-size=' + size,
+        '--agency.wait-for-sync=true',
+        '--agency.supervision=true',
+        '--server.threads=16',
+        '--agency.supervision-frequency=0.5',
+        '--agency.supervision-grace-period=2.5',
+        '--agency.compaction-step-size='+compactionStep,
+        '--agency.compaction-keep-size='+compactionKeep,
+        '--agency.my-address=' + endpoint,
+        instances.length ?
+          `--agency.endpoint=${instances[0].endpoint}` : `--agency.endpoint=${endpoint}`
+      ];
+      this.agentCounter++;
+      debugLog("booting agency");
+      const instance = await this.startArango(
+        'agency-' + this.agentCounter,
+        endpoint,
+        'agent',
+        args
+      );
+      instances.push(instance);
     }
     return this.instances = instances;
   }
 
+  async dumpAgency() {
+    const baseUrl = endpointToUrl(this.getAgencyEndpoint());
+
+    try {
+      let body = await rp({
+        method: 'POST', json: true,
+        uri: `${baseUrl}/_api/agency/read`,
+        followAllRedirects: true,      
+        body: [['/']]
+      });
+      console.error(JSON.stringify(body));
```

Comment:
Is this supposed to be a return? Otherwise the `return null` for the error case doesn't make sense and can be left out. Currently the function returns null on error and undefined on success.